src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -407,8 +407,15 @@ static string GetTypeNameFromCustomNamespace(XmlType xmlType, Compilation compil
 						if (type.ContainingAssembly.Identity.Name != typeInfo.assemblyName)
 							continue;
 
-						if (IsPublicOrVisibleInternal(type, caches.InternalsVisible))
-							return fullName;
+						if (!IsPublicOrVisibleInternal(type, caches.InternalsVisible))
+							continue;
+
+						int i = fullName.IndexOf('`');
+						if (i > 0)
+						{
+							fullName = fullName.Substring(0, i);
+						}
+						return fullName;
```

Comment:
what if it's an open generic type ?

Reply:
So what if it is?  ``OnPlatform`1`` is not valid in C# source; we *must* strip out the type arity.  It needs to either be closed `OnPlatform<string>` or open `OnPlatform<T>`, and never ever never ever ``OnPlatform`1``.

I'm more concerned about *nesting*, e.g. if you wanted to inherit from `Outer<string>.Inner`.  I have no idea how to even express that in XAML -- *can it* be expressed? -- but I'm quite certain that won't work at all right now, because the resulting identifier will contain `/`, and the C# compiler is *not* gonna like inheriting from `Outer/Inner<string>`.

https://github.com/dotnet/maui/blob/e4124ab26caa73a5cb2bb3b5936b743e0cf47305/src/Controls/src/SourceGen/CodeBehindGenerator.cs#L397-L398

Reply:
hopefully, nested types can't be expressed in Xaml

Reply:
well, not as an element syntax

Reply:
@StephaneDelcroix : is it *possible* to specify open type parameters in XAML?  I can't determine the syntax.